packages/core/js-client/src/clientPeer/__test__/client.spec.ts

```diff
@@ -17,15 +17,44 @@
 import { JSONValue } from "@fluencelabs/interfaces";
 import { it, describe, expect } from "vitest";
 
+import { ExpirationError, SendError } from "../../jsPeer/errors.js";
 import { CallServiceData } from "../../jsServiceHost/interfaces.js";
-import { doNothing } from "../../jsServiceHost/serviceUtils.js";
 import { handleTimeout } from "../../particle/Particle.js";
 import { registerHandlersHelper, withClient } from "../../util/testUtils.js";
 import { checkConnection } from "../checkConnection.js";
 
 import { nodes, RELAY } from "./connection.js";
 
 describe("FluenceClient usage test suite", () => {
+  it("Should resolve at TTL when fire and forget behavior is used", async () => {
+    await withClient(RELAY, { defaultTtlMs: 600 }, async (peer) => {
+      const script = `
+    (seq
+        (call %init_peer_id% ("load" "relay") [] init_relay)
+        (call init_relay ("peer" "timeout") [60000 "Do you really want to wait for so long?"])
+    )`;
+
+      const particle = await peer.internals.createNewParticle(script);
+
+      const now = Date.now();
+
+      const promise = new Promise<JSONValue>((resolve, reject) => {
+        registerHandlersHelper(peer, particle, {
+          load: {
+            relay: () => {
+              return peer.getRelayPeerId();
+            },
+          },
+        });
+
+        peer.internals.initiateParticle(particle, resolve, reject);
+      });
+
+      await expect(promise).rejects.toThrow(ExpirationError);
+      expect(Date.now() - 500).toBeGreaterThanOrEqual(now);
```

Comment:
why 500? doc

and some error message for when `expect` fails, so that it's possible to understand why test failed

Reply:
I would suggest to calculate how much time it took, and then put `expect` on that value. That way, code will much easier to understand.

```
const start = Date.now();
...
const elapsed = start - Date.now();
expect(elapsed < 500, "Sending particle took longer than 500ms")
```